src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.epics.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import { ofType } from 'redux-observable';
+import { catchError, flatMap, map, startWith } from 'rxjs/operators';
+import { from, of } from 'rxjs';
+import moment from 'moment';
+import { getApi } from '../../../d2';
+import {
+    enrollmentPageActionTypes,
+    showErrorViewOnEnrollmentPage,
+    showLoadingViewOnEnrollmentPage,
+    successfulFetchingEnrollmentPageInformationFromUrl,
+} from './EnrollmentPage.actions';
+
+const fetchEnrollment = id => getApi().get(`enrollments/${id}`, { fields: 'trackedEntityInstance,program,orgUnit' });
+const fetchTrackedEntityInstance = id => getApi().get(`trackedEntityInstances/${id}`, { fields: 'attributes,enrollments' });
+
+export const fetchEnrollmentPageInformationFromUrlEpic = (action$: InputObservable, store: ReduxStore) =>
+    action$.pipe(
+        ofType(enrollmentPageActionTypes.ENROLLMENT_PAGE_INFORMATION_FETCH),
+        flatMap(() => {
+            const { query: { enrollmentId } } = store.value.router.location;
+
+            return from(fetchEnrollment(enrollmentId))
+                .pipe(
+                    flatMap(({ trackedEntityInstance }) =>
+                        from(fetchTrackedEntityInstance(trackedEntityInstance))
+                            .pipe(
+                                map(({ attributes, enrollments }) => {
+                                    const selectedName = attributes.reduce((acc, { value: dataElementValue }) =>
```

Comment:
We need to use the metadata when computing the name. We should display the attribute value only if the "display in list" flag is set. We should also use the attribute order from the metadata.

Reply:
Okay I was wondering why I can see that display on list flag. Then I thought that it might have been that on the backend they return only the values that are good for display.. 

Okay then you mean that we should be using the data we have stored in memory right?

Reply:
Yes 👍